src/components/validation/resources/http.rs

```diff
@@ -128,22 +224,180 @@ fn spawn_input_http_client(
+    // via an output sender. We accept/collect events until we're told to shutdown.
+
+    // First, we'll build and spawn our HTTP server.
+    let decoder = codec.into_decoder();
+
+    let (_, http_server_shutdown_tx) =
+        spawn_http_server(task_coordinator, &config, move |request| {
+            let output_tx = output_tx.clone();
+            let mut decoder = decoder.clone();
+
+            async move {
+                match hyper::body::to_bytes(request.into_body()).await {
+                    Err(_) => StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+                    Ok(body) => {
+                        let mut body = BytesMut::from(&body[..]);
+                        loop {
+                            match decoder.decode_eof(&mut body) {
+                                Ok(Some((events, _byte_size))) => {
+                                    for event in events {
+                                        output_tx
+                                            .send(event)
+                                            .await
+                                            .expect("should not fail to send output event");
+                                    }
+                                }
+                                Ok(None) => return StatusCode::OK.into_response(),
+                                Err(_) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+                            }
+                        }
+                    }
+                }
+            }
+        });
+
+    // Now we'll create and spawn the resource's core logic loop which simply waits for the runner
+    // to instruct us to shutdown, and when that happens, cascades to shutting down the HTTP server.
+    let resource_started = task_coordinator.track_started();
+    let resource_completed = task_coordinator.track_completed();
+    let mut resource_shutdown_rx = task_coordinator.register_for_shutdown();
+    tokio::spawn(async move {
+        resource_started.mark_as_done();
+        debug!("HTTP server external output resource started.");
+
+        resource_shutdown_rx.wait().await;
+        let _ = http_server_shutdown_tx.send(());
+        resource_completed.mark_as_done();
+
+        debug!("HTTP server external output resource completed.");
+    });
 }
 
+/// Spawns an HTTP client that pulls events by making requests to an HTTP server driven by a sink.
 #[allow(clippy::missing_const_for_fn)]
 fn spawn_output_http_client(
-    _config: HttpConfig,
+    _config: HttpResourceConfig,
     _codec: ResourceCodec,
     _output_tx: mpsc::Sender<Event>,
     _task_coordinator: &TaskCoordinator<Configuring>,
 ) {
+    // TODO: The `prometheus_exporter` sink is the only sink that exposes an HTTP server which must be
+    // scraped... but since we need special logic to aggregate/deduplicate scraped metrics, we can't
+    // use this generically for that purpose.
+    todo!()
+}
+
+fn spawn_http_server<H, F, R>(
```

Comment:
Missing a comment for this function

Reply:
I/we usually only document public functions, though... 🥺

Reply:
Oof, sorry - bad reading comprehension. I thought it was another in the line of documented functions above it. Now I see that it's called _in_ those.